web/lib/HabitatCommunities.js

```diff
@@ -1,61 +1,385 @@
     this._loaded = {};
+    this._userLoaded = {};
+    checkScroll(
+      this.shadowRoot.querySelector('#content'),
+      () => {
+        for (const community of this.communities) {
+          if (!this._loaded[community.transactionHash]) {
+            this._loaded[community.transactionHash] = true;
+            this.renderCommunity(this.container, community);
+          }
+        }
+      }
+    );
+    this._userLoad(
+      () => {
+        for (const community of this.userCommunities) {
+          if (!this._userLoaded[community.transactionHash]) {
+            this._userLoaded[community.transactionHash] = true;
+            this.renderCommunity(this.userContainer, community);
+          }
+        }
+      }
+    );
+
+    this.sortBtn = this.shadowRoot.querySelector('button#sort');
+    this.dropdown = this.shadowRoot.querySelector('#sort-dropdown');
+    this.selector = this.shadowRoot.querySelector('#sort-options');
+    this.sortBtn.addEventListener('click', evt => {
+        evt.stopPropagation();
+        this.dropdown.classList.toggle('active');
+        if (this.dropdown.classList.contains('active')) {
+          this.selector.addEventListener('click', evt => {
+            if (evt.target.id.startsWith('sort-') && !evt.target.id.match('sort-options')) {
+              this.userContainer.innerHTML = '';
+              this._userLoaded = {};
+              this.container.innerHTML = '';
+              this._loaded = {};
+
+              if (evt.target.id === 'sort-az') {
+                this.communities.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()) ? 1 : -1);
+                this.userCommunities.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()) ? 1 : -1);
+              }
+              if (evt.target.id === 'sort-za') {
+                this.communities.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase()) ? 1 : -1);
+                this.userCommunities.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase()) ? 1 : -1);
+              }
+              if (evt.target.id === 'sort-recent') {
+                this.communities.sort((a, b) => (a.block < b.block) ? 1 : -1);
+                this.userCommunities.sort((a, b) => (a.block < b.block) ? 1 : -1);
+              }
+            }
+          }, false);
+          //click anywhere in communities tab to close
+          this.shadowRoot.addEventListener('click', evt => {
+            if (!evt.target.id.startsWith('sort-')) {
+              this.dropdown.classList.remove('active');
+            }
+          }, false);
+        }
+    });
+
   }
 
   get title () {
-    return 'Communites on Habitat';
+    return 'Communites';
   }
 
+  _userLoad (callback) {
+    async function _check () {
+      await callback();
+      window.requestAnimationFrame(_check);
+    }
+    _check();
+  }
+
+  //render elements
   async render () {
+    const { habitat } = await getProviders();
+
+    if (walletIsConnected()) {
```

Comment:
should be moved to onChainUpdate. 
I also not sure if we should hide it by default, this could confuse users looking for a button :)

Reply:
> * missing "create" button in the overview page 
I'm thinking - show "create" button by default & prompt user with wallet connection upon pressing

Reply:
where is onChainUpdate? Should I add to HabitatCommunities.js, edit in HabitatPanel.js, or something else? 
I believe this is the final task I have to complete.

Reply:
Sorry, its called `chainUpdateCallback` in this context